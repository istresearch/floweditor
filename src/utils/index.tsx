import {
  Action,
  Case,
  Category,
  ContactProperties,
  FlowPosition,
  LocalizationMap
} from 'flowTypes';
import { Query } from 'immutability-helper';
import * as React from 'react';
import Localization, { LocalizedObject } from 'services/Localization';
import { Asset, Assets } from 'store/flowContext';
import { FormEntry } from 'store/nodeEditor';
import { v4 as generateUUID } from 'uuid';
import variables from 'variables.module.scss';

export const V4_UUID = /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i;
const LABEL_CHARS = /^[a-zA-Z0-9-][a-zA-Z0-9- ]*$/;
const MAX_LABEL_LENGTH = 36;
const SNAKED_CHARS = /\s|-+(?=\S)/g;

export const MAX_REFLOW_ATTEMPTS = 100;
export const ACTIVITY_INTERVAL = 5000;
export const COLLISION_FUDGE = 5;
export const GRID_SIZE: number = parseInt(variables.grid_size, 10) || 20;
export const NODE_SPACING: number = parseInt(variables.node_spacing, 10) || 10;
export const NODE_PADDING: number = parseInt(variables.node_padding, 10) || 10;
export const QUIET_NOTE = 2000;
export const CONFIRMATION_TIME = 2000;

export interface BoolMap {
  [key: string]: boolean;
}

interface Bounds {
  left: number;
  top: number;
  right: number;
  bottom: number;
}

export type LabelIdCb = (label?: string, labelKey?: string, valueKey?: string) => string;

/**
 * Adjusts the left and top offsets to a grid
 * @param left horizontal offset
 * @param top vertical offset
 */
export const snapToGrid = (left: number, top: number): { left: number; top: number } => {
  let leftAdjust = left % GRID_SIZE;
  let topAdjust = top % GRID_SIZE;

  if (leftAdjust > GRID_SIZE / 3) {
    leftAdjust = GRID_SIZE - leftAdjust;
  } else {
    leftAdjust = leftAdjust * -1;
  }

  if (topAdjust > GRID_SIZE / 3) {
    topAdjust = GRID_SIZE - topAdjust;
  } else {
    topAdjust = topAdjust * -1;
  }

  return {
    left: Math.max(left + leftAdjust, 0),
    top: Math.max(top + topAdjust, 0)
  };
};

/**
 * Adjusts the position offsets to a grid
 */
export const snapPositionToGrid = (position: FlowPosition): FlowPosition => {
  let leftAdjust = position.left % GRID_SIZE;
  let topAdjust = position.top % GRID_SIZE;

  if (leftAdjust > GRID_SIZE / 3) {
    leftAdjust = GRID_SIZE - leftAdjust;
  } else {
    leftAdjust = leftAdjust * -1;
  }

  if (topAdjust > GRID_SIZE / 3) {
    topAdjust = GRID_SIZE - topAdjust;
  } else {
    topAdjust = topAdjust * -1;
  }

  const left = Math.max(position.left + leftAdjust, 0);
  const top = Math.max(position.top + topAdjust, 0);
  const right = left + position.right - position.left;
  const bottom = top + position.bottom - position.top;

  return {
    left,
    top,
    right,
    bottom
  };
};

/**
 * Turns a string array into a bool map for constant lookup
 * @param {string[]} array - an array of strings, e.g. contact fields
 * @returns {object} A map of each string
 */
export const toBoolMap = (array: string[]): BoolMap =>
  array.reduce(
    (map: BoolMap, item: string) => ({
      ...map,
      [item]: true
    }),
    {}
  );

/**
 * Inserts commas into numbers where appropriate for better readability
 * @param {number} value - A number, e.g. 10000
 * @returns {string} A comma-separated string, e.g. 10,000
 */
export const addCommas = (value: number): string =>
  value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');

/**
 * Replaces spaces with underscores
 * @param {string} value - A space-separated string to be snaked, e.g. a raw flow field name ('my flow field')
 * @returns {string} A snaked string, e.g. 'my_flow_field'
 */
export const snakify = (value: string): string =>
  value
    .toLowerCase()
    .trim()
    .replace(SNAKED_CHARS, '_');
/**
 * Returns true if a given UUID matches v4 format
 * @param {string} uuid - A version 4 UUID (no braces, uppercase OK)
 * @returns {boolean}
 */
export const validUUID = (uuid: string): boolean => V4_UUID.test(uuid);

/**
 * Returns a given string in title case, e.g. 'full name' becomes 'Full Name'
 * @param {string} str - string to be title-cased
 * @returns {string} Title-cased string
 */
export const titleCase = (str: string): string =>
  str.replace(/\b\w+/g, s => s.charAt(0).toUpperCase() + s.substr(1).toLowerCase());

export const getSelectClassForEntry = (entry: FormEntry): string => {
  if (entry && entry.validationFailures && entry.validationFailures.length > 0) {
    return 'select-invalid';
  }
  return '';
};

export const getSelectClass = (errors: number): string => {
  if (errors === 0) {
    return 'react-select select-base';
  }
  // We use a global selector here for react-select
  return 'react-select select-base select-invalid';
};

export const reorderList = (list: any[], startIndex: number, endIndex: number): any[] => {
  const [removed] = list.splice(startIndex, 1);

  list.splice(endIndex, 0, removed);

  return list;
};

/**
 * Compares basic objects (no methods and DOM nodes; property order important)
 * @param {object} objA - basic object
 * @param {object} objB - basic object
 * @returns {boolean}
 */
export const jsonEqual = (objA: {}, objB: {}): boolean =>
  JSON.stringify(objA) === JSON.stringify(objB);

/**
 * Checks whether any of a list of error strings contain one or more queries.
 * Used to determine whether a particular error has been encountered.
 */
export const hasErrorType = (errors: string[], exps: RegExp[]): boolean => {
  if (!errors.length) {
    return false;
  }
  for (const error of errors) {
    for (const exp of exps) {
      if (error.match(exp)) {
        return true;
      }
    }
  }
  return false;
};

export const getLocalizedObject = (localizations: LocalizedObject[]) => {
  if (localizations && localizations.length) {
    return localizations[0];
  }
};

export interface ClickHandler {
  // onClick: (event: React.MouseEvent<HTMLDivElement>) => void;
  onMouseDown: (event: React.MouseEvent<HTMLDivElement>) => void;
  onMouseUp: (event: React.MouseEvent<HTMLDivElement>) => void;
}

/**
 * Creates a simple click handler via onMouseDown and onMouseUp.
 * This is a necessity in order to let jsPlumb manage our element dragging.
 * @param onClick
 */
export const createClickHandler = (
  onClick: (event: React.MouseEvent<HTMLElement>) => void,
  shouldCancelClick: () => boolean = null,
  onMouseDown?: (event: React.MouseEvent<HTMLElement>) => void
): any => {
  return {
    onMouseDown: (event: React.MouseEvent<HTMLElement>) => {
      if (onMouseDown) {
        onMouseDown(event);
      }
    },
    onMouseUp: (event: React.MouseEvent<HTMLElement>) => {
      if (!shouldCancelClick || !shouldCancelClick()) {
        onClick(event);
      }
    }
  };
};

export const getLocalization = (
  obj: Action | Category | Case,
  localization: LocalizationMap,
  language: Asset
) => Localization.translate(obj, language, localization[language.id]);

/** istanbul ignore next */
export const dump = (thing: any) => console.log(JSON.stringify(thing, null, 4));

/**
 * Apply emphasis style
 */
export const emphasize = (text: string) => <span className="emphasize">{text}</span>;
export const ellipsize = (text: string, max: number = 250) => {
  if (text.length > max) {
    return text.substr(0, max) + '...';
  }
  return text;
};

/**
 * Does property exist in our ContactProperties enum?
 * @param {string} propertyToCheck - 'name' key on react-select AutoCompleteOption
 */
export const propertyExists = (propertyToCheck: string) => {
  for (const property of Object.keys(ContactProperties)) {
    if (property.toLowerCase() === propertyToCheck.toLowerCase().trim()) {
      return true;
    }
  }
  return false;
};

/**
 * Should x element be rendered?
 */
export const renderIf = (predicate: boolean) => (then: JSX.Element, otherwise?: JSX.Element) =>
  predicate ? then : otherwise ? otherwise : null;

/**
 * Does the label meet our length requirements?
 * @param {string} label - label created by react-select
 */
export const properLabelLength = (label: string = '') =>
  label.length > 0 && label.length <= MAX_LABEL_LENGTH;

/**
 * Does the label meet our character requirements?
 * @param {string} label
 */
export const containsOnlyLabelChars = (label: string = '') => LABEL_CHARS.test(label);

/**
 * Does the label meet our length, character requirements?
 * @param {string} label - label created by react-select
 */
export const isValidLabel = (label: string) =>
  properLabelLength(label) && containsOnlyLabelChars(label);

export const isRealValue = (obj: any) => obj !== null && obj !== undefined;

/* istanbul ignore next */
export const timeStart = (name: string) =>
  process.env.NODE_ENV === 'development' && console.time(name);

/* istanbul ignore next */
export const timeEnd = (name: string) =>
  process.env.NODE_ENV === 'development' && console.timeEnd(name);

export const log = (...message: any[]) =>
  process.env.NODE_ENV === 'development' && console.log(...message);

export const capitalize = (str: string) =>
  str.replace(/(?:^|\s)\S/g, captured => captured.toUpperCase());

export const set = (val: any): Query<any> => ({ $set: val });

export const setTrue = (): Query<true> => set(true);

export const setFalse = (): Query<false> => set(false);

export const setNull = (): Query<null> => set(null);

export const setEmpty = (): Query<string> => set('');

export const merge = (val: any): Query<any> => ({ $merge: val });

export const unset = (val: any): Query<any> => ({ $unset: val });

export const push = (arr: any[]): Query<any[]> => ({ $push: arr });

// tslint:disable-next-line:array-type
export const splice = (arr: Array<Array<any>>): Query<Array<Array<any>>> => ({
  $splice: arr
});

export const optionExists = (newOptName: string, options: any[]) =>
  options.find(({ name }) => name.toLowerCase().trim() === newOptName.toLowerCase().trim())
    ? true
    : false;

/* export const isOptionUnique: IsOptionUniqueHandler = ({ option, options, labelKey, valueKey }) =>
    !propertyExists(option.name) && !optionExists(option.name, options);

export const isValidNewOption: IsValidNewOptionHandler = ({ label }) => isValidLabel(label);
*/
/* export const composeCreateNewOption = ({
    idCb,
    type
}: {
    idCb: LabelIdCb;
    type: AssetType;
}): NewOptionCreatorHandler => ({ label, labelKey, valueKey }) => ({
    id: idCb(label, labelKey, valueKey),
    name: label,
    type,
    isNew: true
});*/

/**
 * Deduplicate values in an array.
 * Pass it a key to deduplicate based on obj key.
 */
export const uniqueBy = (a: any[], key: string): any[] => {
  const seen: any = {};
  return a.filter((item: any) => {
    const k = item[key];
    return seen.hasOwnProperty(k) ? false : (seen[k] = true);
  });
};

export const downloadJSON = (obj: any, name: string): void => {
  const dataStr =
    'data:text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(obj, null, 2));
  const downloadAnchorNode = document.createElement('a');
  downloadAnchorNode.setAttribute('href', dataStr);
  downloadAnchorNode.setAttribute('download', name + '.json');
  document.body.appendChild(downloadAnchorNode); // required for firefox
  downloadAnchorNode.click();
  downloadAnchorNode.remove();
};

export const fetchAsset = (asset: Asset) => (assets: Assets, id: string): Promise<Asset> => {
  return new Promise<Asset>((resolve, reject) => {
    resolve(asset);
  });
};

export let createUUID = (): string => {
  return generateUUID();
};

export const seededUUIDs = (seed: number = 1): any => {
  let current = seed;
  const randomNext = () => {
    const x = Math.sin(current++) * 10000;
    return x - Math.floor(x);
  };

  const random = (min: number, max: number): number => {
    return Math.floor(randomNext() * (max - min) + min);
  };

  return () => {
    const randomSeed: number[] = [];
    for (let idx = 0; idx < 16; idx++) {
      randomSeed.push(random(0, 250));
    }
    return generateUUID({ random: randomSeed });
  };
};

export const range = (start: number, end: number) =>
  Array.from({ length: end - start }, (v: number, k: number) => k + start);

export const pluralize = (count: number, noun: string, suffix: string = 's'): string =>
  `${noun}${count !== 1 ? suffix : ''}`;

export const hasString = (names: string[], key: string): boolean =>
  !!names.find((item: string) => item === key);

export const scalarArrayEquals = (a: any[], b: any[]) =>
  a.every((value, index) => value === b[index]);

export const getURNPath = (urn: string) => {
  return urn.split(':')[1];
};

export const copyToClipboard = (text: string) => {
  navigator.clipboard.writeText(text);
};

export const throttle = (func: any, timeout: any) => {
  let ready: boolean = true;

  return (...args: any) => {
    if (!ready) {
      return;
    }

    ready = false;
    func(...args);
    setTimeout(() => {
      ready = true;
    }, timeout);
  };
};

export const traceUpdate = (component: any, prevProps: any, prevState?: any) => {
  const messages: string[] = [];
  Object.entries(component.props).forEach(
    ([key, val]) => prevProps[key] !== val && messages.push(`Prop: '${key}' changed`)
  );
  if (prevState && component.state) {
    Object.entries(component.state).forEach(
      ([key, val]) => prevState[key] !== val && messages.push(`State: '${key}' changed`)
    );
  }

  if (messages.length > 0) {
    log('****  ' + component._reactInternalFiber.type.name + ' changed');
    messages.forEach((message: string) => {
      log(message);
    });
  }
};

export const debounce = (fn: any, quiet: number, closure: any = null) => {
  if (fn.timeout) {
    window.clearTimeout(fn.timeout);
  }
  fn.timeout = window.setTimeout(closure || fn, quiet);
};

export const onNextRender = (fn: any) => {
  window.setTimeout(fn, 0);
};

export const desnake = (text: string): string => {
  if (!text) {
    return text;
  }
  return text
    .split('_')
    .join(' ')
    .toLowerCase();
};

export const bool = (prop: boolean) => {
  return prop ? 'true' : null;
};
