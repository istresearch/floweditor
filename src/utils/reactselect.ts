import { StylesConfig } from 'react-select';

export const large = {
  placeholder: (styles: StylesConfig, state: any) => {
    return { ...styles, color: '#ddd', marginLeft: '3px' };
  },
  singleValue: (styles: StylesConfig, state: any) => {
    return { ...styles, marginLeft: '3px' };
  },
  multiValue: (styles: StylesConfig, state: any) => {
    return { ...styles, marginLeft: '0px' };
  },
  input: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      marginLeft: '0px',
      caretColor: '#999',
      marginBottom: '0px',
      boxShadow: 'none',
      border: 'none',
      fontSize: '13px'
    };
  },
  control: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      borderColor: '#e6e6e6',
      boxShadow: 'none',
      fontSize: '13px',
      transition: 'all 0.3s ease-in-out',
      '&:hover': {
        borderColor: '#e6e6e6'
      },
      '&:focus-within': {
        boxShadow: 'var(--widget-box-shadow-focused) !important',
        border: '1px solid #a4cafe !important'
      }
    };
  }
};

export const messageStyle = {
  control: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      borderColor: '#e6e6e6',
      boxShadow: 'none',
      borderBottom: 'none',
      borderBottomRightRadius: '0px',
      borderBottomLeftRadius: '0px',
      '&:hover': {
        borderColor: '#e6e6e6'
      }
    };
  }
};

export const getErroredSelect = (baseControl: any): any => {
  return {
    control: (styles: StylesConfig) => ({
      ...styles,
      ...baseControl,
      borderColor: 'tomato',
      boxShadow: '0 0 0 3px rgba(255,196,186,0.5) !important;',
      '&:hover': {
        borderColor: 'tomato'
      }
    })
  };
};

export const small = {
  control: (styles: StylesConfig) => ({
    ...styles,
    borderColor: '#e6e6e6',
    boxShadow: 'none',
    '&:hover': {
      borderColor: '#e6e6e6'
    },
    maxHeight: '28px',
    minHeight: '28px',
    fontSize: '12px',
    lineHeight: '12px',
    marginBottom: '-10px',
    '&:focus-within': {
      boxShadow: 'var(--widget-box-shadow-focused) !important',
      border: '1px solid #a4cafe !important'
    }
  }),
  menuList: (styles: StylesConfig) => ({
    ...styles,
    maxHeight: 200
  }),
  option: (styles: StylesConfig) => ({
    ...styles,
    fontSize: 12
  }),
  singleValue: (styles: StylesConfig) => ({
    ...styles,
    maxHeight: 28,
    lineHeight: '13px'
  }),
  indicatorsContainer: (styles: StylesConfig) => ({
    ...styles,
    height: 28,
    maxHeight: 28
  }),
  valueContainer: (styles: StylesConfig) => ({
    ...styles,
    height: 28,
    maxHeight: 28
  }),
  input: (styles: StylesConfig) => ({
    ...styles,
    marginLeft: '-2px',
    marginTop: '0px',
    caretColor: '#999',
    boxShadow: 'none',
    border: 'none',
    maxHeight: '20px',
    minHeight: '20px',
    lineHeight: '12px'
  })
};

export const tagging = {
  control: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      borderColor: '#e6e6e6',
      boxShadow: 'none',
      transition: 'all 0.3s ease-in-out',
      '&:hover': {
        borderColor: '#e6e6e6'
      },
      '&:focus-within': {
        boxShadow: 'var(--widget-box-shadow-focused) !important',
        border: '1px solid #a4cafe !important'
      }
    };
  },
  input: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      marginLeft: '0px',
      caretColor: '#999',
      marginBottom: '0px',
      boxShadow: 'none',
      border: 'none'
    };
  },
  multiValue: (styles: StylesConfig) => {
    return {
      ...styles,
      maxHeight: 24,
      minHeight: 24,
      lineHeight: '17px',
      backgroundColor: 'hsl(0, 0%, 94%)',
      border: '1px solid hsl(0, 0%, 85%)'
    };
  }
};
