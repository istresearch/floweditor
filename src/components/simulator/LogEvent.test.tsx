import { render, mock, fireEvent } from 'test/utils';
import * as React from 'react';
import LogEvent, { EventProps } from './LogEvent';

const commonEventProps: EventProps = {
  uuid: '51e6f864-a16f-4be7-839f-945afc857559',
  step_uuid: '6de81ff6-f541-4099-8ad7-03214d15b07d',
  created_on: '2020-01-29T10:43:30.123456789Z'
};

const testEventRender = (eventProps: EventProps) => {
  const props: EventProps = {
    ...eventProps
  };
  const { baseElement } = render(<LogEvent {...props} />);

  expect(baseElement).toMatchSnapshot();
};

describe(LogEvent.name, () => {
  it('should render broadcast_created event', () => {
    testEventRender({
      type: 'broadcast_created',
      groups: [{ uuid: '3a3e061e-dad5-4454-88e4-ccbd0ef0e475', name: 'U-Reporters' }],
      translations: {
        eng: {
          text: 'Hi there'
        },
        spa: {
          text: 'Hola'
        }
      },
      base_language: 'eng',
      ...commonEventProps
    });
  });
  it('should render contact_field_changed event', () => {
    testEventRender({
      type: 'contact_field_changed',
      field: { key: 'age', name: 'Age' },
      value: { text: '38' },
      ...commonEventProps
    });
  });
  it('should render contact_field_changed event with null value', () => {
    testEventRender({
      type: 'contact_field_changed',
      field: { key: 'age', name: 'Age' },
      value: null,
      ...commonEventProps
    });
  });
  it('should render contact_groups_changed event', () => {
    testEventRender({
      type: 'contact_groups_changed',
      groups_added: [
        { uuid: '3a3e061e-dad5-4454-88e4-ccbd0ef0e475', name: 'U-Reporters' },
        { uuid: 'd749781c-bb3a-4894-81fb-683b6368b29c', name: 'Youth' }
      ],
      groups_removed: [
        { uuid: '669ce0aa-0444-4597-87a0-feb82401a31d', name: 'Unregistered' },
        { uuid: '58d99177-15f4-4a25-9b35-222e09252387', name: 'Missing Name' }
      ],
      ...commonEventProps
    });
  });
  it('should render contact_name_changed event', () => {
    testEventRender({
      type: 'contact_name_changed',
      name: 'Bobby',
      ...commonEventProps
    });
  });
  it('should render contact_language_changed event', () => {
    testEventRender({
      type: 'contact_language_changed',
      language: 'eng',
      ...commonEventProps
    });
  });
  it('should render contact_status_changed event', () => {
    testEventRender({
      type: 'contact_status_changed',
      status: 'blocked',
      ...commonEventProps
    });
  });
  it('should render contact_urns_changed event', () => {
    testEventRender({
      type: 'contact_urns_changed',
      urns: ['tel:+1234567890', 'twitter:bobby'],
      ...commonEventProps
    });
  });
  it('should render email_sent event', () => {
    testEventRender({
      type: 'email_sent',
      subject: 'Party time',
      body: 'Dear Sir/Madam',
      to: ['fun@temba.io', 'events@temba.io'],
      ...commonEventProps
    });
  });
  it('should render error event', () => {
    testEventRender({
      type: 'error',
      text: "I'm an error",
      ...commonEventProps
    });
  });
  it('should render failure event', () => {
    testEventRender({
      type: 'failure',
      text: "I'm a failure",
      ...commonEventProps
    });
  });
  it('should render input_labels_added event', () => {
    testEventRender({
      type: 'input_labels_added',
      labels: [
        { uuid: '3a3e061e-dad5-4454-88e4-ccbd0ef0e475', name: 'Spam' },
        { uuid: 'd749781c-bb3a-4894-81fb-683b6368b29c', name: 'Important' }
      ],
      ...commonEventProps
    });
  });
  it('should render ivr_created event', () => {
    testEventRender({
      type: 'ivr_created',
      msg: {
        uuid: 'c166c2cb-290c-4805-a5af-052ad2858288',
        urn: 'tel:+1123456789',
        text: 'Thanks for getting in touch',
        attachments: ['image/jpeg:http://temba.io/test.wav']
      },
      ...commonEventProps
    });
  });
  it('should render msg_created event', () => {
    testEventRender({
      type: 'msg_created',
      msg: {
        uuid: 'c166c2cb-290c-4805-a5af-052ad2858288',
        urn: 'tel:+1123456789',
        text: 'Hi there',
        attachments: ['image/jpeg:http://temba.io/test.jpg']
      },
      ...commonEventProps
    });
  });
  it('should render msg_received event', () => {
    testEventRender({
      type: 'msg_received',
      msg: {
        uuid: 'c166c2cb-290c-4805-a5af-052ad2858288',
        urn: 'tel:+1123456789',
        text: 'Thanks for getting in touch',
        attachments: ['image/jpeg:http://temba.io/test.jpg']
      },
      ...commonEventProps
    });
  });
  it('should render msg_wait event', () => {
    testEventRender({
      type: 'msg_wait',
      ...commonEventProps
    });
  });
  it('should render resthook_called event', () => {
    testEventRender({
      type: 'resthook_called',
      resthook: 'new-registration',
      ...commonEventProps
    });
  });
  it('should render run_result_changed event', () => {
    testEventRender({
      type: 'run_result_changed',
      name: 'Response 1',
      value: 'yes',
      ...commonEventProps
    });
  });
  it('should render service_called event', () => {
    testEventRender({
      type: 'service_called',
      service: 'classifier',
      classifier: { uuid: 'a67d9cc0-15bb-4a92-b387-554e28726472', name: 'Booking (Wit)' },
      http_logs: [
        {
          url: 'https://api.wit.ai/message?v=20170307&q=Hi+everybody',
          request:
            'GET /message?v=20170307&q=Hi+everybody HTTP/1.1\r\nHost: api.wit.ai\r\nUser-Agent: Go-http-client/1.1\r\nAuthorization: Bearer 123456789\r\nAccept-Encoding: gzip\r\n\r\n',
          response:
            'HTTP/1.0 200 OK\r\nContent-Length: 127\r\n\r\n{"_text":"Hi everyone","entities":{"intent":[{"confidence":0.84709152161066,"value":"greeting"}]},"msg_id":"1M7fAcDWag76OmgDI"}'
        }
      ],
      ...commonEventProps
    });
  });
  it('should render ticket_opened event', () => {
    testEventRender({
      type: 'ticket_opened',
      ticketer: { uuid: '15892014-144c-4721-a611-c80b38481055', name: 'Email Support' },
      ticket: {
        topic: { uuid: 'ee1e36a8-dd42-4464-b6b2-37418a26db1f', name: 'Support' },
        body: 'Where are my cookies?'
      },
      result_name: 'Ticket',
      ...commonEventProps
    });
  });
});
