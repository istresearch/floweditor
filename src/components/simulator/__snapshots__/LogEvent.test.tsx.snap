// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogEvent should render broadcast_created event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="send_msg"
      >
        <div
          class="msg_text"
        >
          Hi there
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render contact_field_changed event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Set contact "Age" to "38"
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render contact_field_changed event with null value 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Cleared contact "Age"
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render contact_groups_changed event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Added to "U-Reporters", "Youth". Removed from "Unregistered", "Missing Name"
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render contact_language_changed event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Set preferred language to "eng"
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render contact_name_changed event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Set contact name to "Bobby"
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render contact_status_changed event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Set status to "blocked"
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render contact_urns_changed event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Added a URN for the contact
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render email_sent event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div>
        <div
          class="has_detail"
        >
          <div
            class="info email"
          >
            Sent email to &quot;fun@temba.io&quot;, &quot;events@temba.io&quot; with subject "Party time"
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;

exports[`LogEvent should render error event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="error"
      >
        <span>
          Error: 
          I'm an error
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render failure event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="error"
      >
        <span>
          Error: 
          I'm a failure
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render input_labels_added event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Message labeled with "Spam", "Important"
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render ivr_created event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="send_msg"
      >
        <div>
          <img
            alt="Attachment"
            src="http://temba.io/test.wav"
          />
        </div>
        <div
          class="msg_text"
        >
          Thanks for getting in touch
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render msg_created event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="send_msg"
      >
        <div>
          <img
            alt="Attachment"
            src="http://temba.io/test.jpg"
          />
        </div>
        <div
          class="msg_text"
        >
          Hi there
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render msg_received event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="msg_received"
      >
        <div>
          <img
            alt="Attachment"
            src="http://temba.io/test.jpg"
          />
        </div>
        <div
          class="msg_text"
        >
          Thanks for getting in touch
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render msg_wait event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Waiting for reply
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render resthook_called event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Triggered flow event "new-registration"
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render run_result_changed event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Set result "Response 1" to "yes"
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`LogEvent should render service_called event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div>
        <div
          class="has_detail"
        >
          <div
            class="info webhook"
          >
            <span>
              Called 
              https://api.wit.ai/message?v=20170307&q=Hi+everybody
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;

exports[`LogEvent should render ticket_opened event 1`] = `
<body>
  <div>
    <div
      class="evt"
    >
      <div
        class="info"
      >
        <span>
          Ticket opened with subject "Need help"
        </span>
      </div>
    </div>
  </div>
</body>
`;
